Guide/src/dev_guide/dev_tools/xflowey.md

```diff
@@ -10,13 +10,615 @@ for writing maintainable, cross-platform automation.
+- `cargo xflowey regen` - Compiles the code **and** runs the full build-time resolution to construct the DAG, validate the pipeline, and regenerate all YAML files defined in `.flowey.toml`.
+
+Always run `cargo xflowey regen` after modifying pipeline definitions to ensure the generated YAML files reflect your changes.
+```
+
+This separation allows flowey to:
+- Validate the entire workflow before execution
+- Generate static YAML for CI systems (ADO, GitHub Actions)
+- Optimize step ordering and parallelization
+- Catch dependency errors at build-time rather than runtime
+
+### Backend Abstraction
+
+Flowey supports multiple execution backends:
+
+- **Local**: Runs directly on your development machine via bash or direct
+  execution
+- **ADO (Azure DevOps)**: Generates ADO Pipeline YAML
+- **GitHub Actions**: Generates GitHub Actions workflow YAML
+
+```admonish warning: 
+Nodes should be written to work across ALL backends whenever possible. Relying on `ctx.backend()` to query the backend or manually emitting 
+backend-specific steps (via `emit_ado_step` or `emit_gh_step`) should be 
+avoided unless absolutely necessary. Most automation logic should be 
+backend-agnostic, using `emit_rust_step` for cross-platform Rust code that 
+works everywhere.
+```
+---
+
+## Pipelines
+
+A **Pipeline** is the top-level construct that defines a complete automation
+workflow. Pipelines consist of one or more **Jobs**, each of which runs a set
+of **Nodes** to accomplish specific tasks.
+
+For detailed examples of defining pipelines, see the [IntoPipeline trait documentation](https://docs.rs/flowey_core/latest/flowey_core/pipeline/trait.IntoPipeline.html).
+
+### Pipeline Jobs
+
+Each `PipelineJob` represents a unit of work that:
+- Runs on a specific platform and architecture
+- Can depend on artifacts from other jobs
+- Can be conditionally executed based on parameters
+- Emits a sequence of steps that accomplish the job's goals
+
+Jobs are configured using a builder pattern:
+
+```rust
+let job = pipeline
+    .new_job(platform, arch, "my-job")
+    .with_timeout_in_minutes(60)
+    .with_condition(some_param)
+    .ado_set_pool("my-pool")
+    .gh_set_pool(GhRunner::UbuntuLatest)
+    .dep_on(|ctx| {
+        // Define what nodes this job depends on
+        some_node::Request { /* ... */ }
+    })
+    .finish();
+```
+
+### Pipeline Parameters
+
+Parameters allow runtime configuration of pipelines:
+
+```rust
+// Define a boolean parameter
+let use_cache = pipeline.new_parameter_bool(
+    "use_cache",
+    "Whether to use caching",
+    ParameterKind::Stable,
+    Some(true) // default value
+);
+
+// Use the parameter in a job
+let job = pipeline.new_job(...)
+    .dep_on(|ctx| {
+        let use_cache = ctx.use_parameter(use_cache);
+        // use_cache is now a ReadVar<bool>
+    })
+    .finish();
+```
+
+Parameter types:
+- Boolean parameters
+- String parameters with optional validation
+- Numeric (i64) parameters with optional validation
+
```

Comment:
prob worth showing off how these params are used in practice across various backends (i.e: how a param maps to a check-box / text field when kicking off manual ADO runs, etc...)